### PR TITLE
SOX-316 Add WS reset password to component WS also

### PR DESCRIPTION
**Pull request by jirkasemmler**
Jira: SOX-316
KBC: XXX

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

**Comment by jirkasemmler** [comment]
je to copypaste z toho puvodni PR

File: src/Keboola/StorageApi/Components.php [hunk 1]
```diff
@@ -329,18 +329,47 @@ public function createConfigurationRowFromVersion($componentId, $configurationId
...
-        return $this->client->apiPostJson(
+        $result = $this->client->apiPostJson(
             $url,
             $options,
             true,
             [Client::REQUEST_OPTION_EXTENDED_TIMEOUT => true]
         );
+        assert(is_array($result));
+
+        $response = $this->resetWorkspacePassword($result['id']);
```
Comment (tomasfejfar) [inline src/Keboola/StorageApi/Components.php:357]: Nemůžeš si udělat new Workspaces($this->client) a volat to na tom?

**Review by tomasfejfar — CHANGES_REQUESTED** [review]

File: src/Keboola/StorageApi/Components.php (hunk 1 above)
Comment (jirkasemmler) [reply src/Keboola/StorageApi/Components.php]: mas pravdu, opraveno

File: src/Keboola/StorageApi/Components.php [hunk 2]
```diff
@@ -335,12 +335,17 @@ public function createConfigurationWorkspace($componentId, $configurationId, arr
...
             $url,
             $options,
             true,
             [Client::REQUEST_OPTION_EXTENDED_TIMEOUT => true]
         );
+        assert(is_array($workspaceResponse));
+
+        $ws = new Workspaces($this->client);
+        $resetPasswordResponse = $ws->resetWorkspacePassword($workspaceResponse['id']);
+        return Workspaces::addCredentialsToWorkspaceResponse($workspaceResponse, $resetPasswordResponse);
```
Comment (tomasfejfar) [inline src/Keboola/StorageApi/Components.php:348]: ❤️

**Review by tomasfejfar — APPROVED** [review]

**Pull request merged**